pkg/gen/typegen.go

```diff
@@ -301,7 +300,18 @@ func makeTypescriptType(prop map[string]interface{}, opts groupOpts) string {
 					}
 				}
 			}
+		} else if tstr == "string" && resourceType == "io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta" && propName == "namespace" {
+			// Special case: `.metadata.namespace` should either take a string or a namespace object
```

Comment:
Is there anything else you can think of that ought to behave like this? If so, is it worth doing something a bit more general-purpose (ala the `pulumi-terraform` `AltTypes` approach)?

Reply:
I actually can, but as they are vastly more specific, we had planned to deal with these in kx instead.

Arguably, this should have gone into kx instead, too, but it's common enough that I'm willing to break from our "pure k8s API" stance.